### PR TITLE
[Doc] MLRun overview & basic-components doc edits

### PR DESCRIPTION
**Pull request by Sharon-iguazio**
Changes include a doc review for recent MLRun-overview product edits in PR #645 (Gilad) and integrate product doc edits in the MLRun getting-started tutorial demo (Adi) with my doc-review edits.
@gilad-shaham @gaydailan please approve. You can see the modified README on the PR branch: https://github.com/Sharon-iguazio/mlrun/blob/doc-readme-edits/README.md

This PR also includes the typo fix in PR #666. => If this PR is merged before PR #666, the latter can be canceled. => UPDATE: PR #666 has now been merged to `development`.

### CODE REVIEW

File: docs/index.rst [hunk 1]
```diff
@@ -11,32 +11,32 @@ MLRun Package Documentation
...
-- **Central data & metadata management** – Log all data, models and artifacts and track all code execution.
-
+- **Feature store** &mdash; define and reuse features with a robust feature store that includes a highly flexible transformation framework.
+- **Elastic serverless runtimes** &mdash; use a single command to turn your Python code into composable functions that can run at scale on Kubernetes, Dask, and Horovod.
+- **Function marketplace** &mdash; leverage a function marketplace to accelerate your model-development process.
+- **Data ingestion & preparation** &mdash; read and transform data from batch and online data stores.
+- **Model training & testing** &mdash; train models at scale with automated testing functions.
+- **Real-time Data & model Pipeline** &mdash; deploy real-time pipelines for data collection and model serving and monitoring.
+- **Data & model Monitoring** &mdash; automate model monitoring and drift detection.
+- **Central data & metadata management** &mdash; log all data, models and artifacts and track all code execution.
```
Comment (gilad-shaham) [inline docs/index.rst:39]: Last I checked &mdash; doesn't work in rst. That's why I put an actual mdash character

**Review by gilad-shaham — CHANGES_REQUESTED** [review]
Looks good, just the &mdash; issue

File: docs/index.rst (hunk 1 above)
Comment (Sharon-iguazio) [reply docs/index.rst]: @gilad-shaham OK, I reverted to using an em-dash Unicode character (—) instead of `&mdash;` in **docs/index.rst**.

**Review by gilad-shaham — APPROVED** [review]

File: docs/index.rst (hunk 1 above)
Comment (Sharon-iguazio) [reply docs/index.rst]: Gilad approved the PR. => I'm resolving this conversation.

**Pull request merged**